upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template

```
@@ -329,6 +329,9 @@ spec:
             - --scale-down-delay-after-add={{ .ScaleDownDelayAfterAdd }}
             - --new-pod-scale-up-delay={{ .NewPodScaleUpDelay }}
             - --max-node-provision-time={{ .MaxNodeProvisionTime }}
+            {{ if .CordonNodeBeforeTerminating }}
+            - --cordon-node-before-terminating=true
+            {{ end }}
```

```suggestion
            # This flag does not exist before CAS 1.21
            {{ IsKubernetesGTE 1.21 }}
             - --cordon-node-before-terminating="{{ WithDefaultBool .CordonNodeBeforeTerminating false }}"
            {{ end }}
```

Although I am wondering if not we could default to true. Seems like the right thing to do.

---

This was introduced on some version of the auto scaler (I don't know exactly, but it was 1.19.x), wouldn't it break compatibility?

---

1.21 is the earliest version. Hence the GTE 1.21 check.